utcp.c

```diff
@@ -26,9 +26,14 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <string.h>
+
+#ifdef _WIN32
+#include <winsock2.h>
+#else
```

Comment:
Weiter unten nutzt du deine in `compat.h` definierten Makros. Also ist die hier included und somit bereits auch schon `winsock2.h`. Hier also bitte wieder rausnehmen.